lib/fizzy/parser.cpp

```diff
@@ -373,13 +373,18 @@ inline Code parse_code(code_view code_binary, FuncIdx func_idx, const Module& mo
             throw parser_error{"too many local variables"};
     }
 
-    const auto [code, pos2] =
-        parse_expr(pos1, end, static_cast<uint32_t>(local_count), func_idx, module);
+    // TODO: Clarify in spec what happens if count of locals and arguments exceed uint32_t::max()
```

Comment:
I'm moving this assertion from `parse_expr` to `parse_code`, because now `parser_expr` doesn't get `local_count` and iterates `locals_vec` anyway on each local access, to find the type.